### PR TITLE
refactor(bus): provide a meaningful error description to the client

### LINKED ISSUE #60: Provide a meaningful error description to the client
It's nice to have a detailed description of the error that the server could provide while processing the request through a response content if any, instead of the standard .net error descriptions based on the response code.

### PR DESCRIPTION
**Pull request by ostridm**
closes #60

### CODE REVIEW

File: src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  public static HttpResponseMessage VerifySuccessStatusCode(this HttpResponseMessage httpResponseMessage)
+  {
+    if (httpResponseMessage.CanHandle())
+    {
+      var message = httpResponseMessage.TryExtractMessage();
+
+      if (message is not null)
+      {
+        throw new HttpRequestException(
+          $"{message}: {(int)httpResponseMessage.StatusCode} ({httpResponseMessage.ReasonPhrase}).");
```
Comment (derevnjuk) [inline src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs]: I can hardly call this error message "meaningful". You have used a ton of cryptic symbols, including an HTTP status code

File: src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    return httpResponseMessage.EnsureSuccessStatusCode();
+  }
+
+  private static bool CanHandle(this HttpResponseMessage httpResponseMessage)
+  {
+    return (int)httpResponseMessage.StatusCode >= 400 &&
+           httpResponseMessage.Content.Headers.ContentType is not null &&
+           httpResponseMessage.Content.Headers.ContentLength is > 0 and < ContentLengthThreshold &&
+           DetailsContentType.Equals(httpResponseMessage.Content.Headers.ContentType.MediaType,
+             StringComparison.OrdinalIgnoreCase);
```
Comment (derevnjuk) [inline src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs]: So, does it means you are not going to handle `text/plain`? It is not clear why don't use `IsSuccessStatusCode`. We have already discussed that we don't expect any other content-type

File: src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  private static string? TryExtractMessage(this HttpResponseMessage httpResponseMessage)
+  {
+    var content = Task.Run(() => httpResponseMessage.Content.ReadAsStringAsync())
+      .ConfigureAwait(false).GetAwaiter().GetResult();
```
Comment (derevnjuk) [inline src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs]: This is the worst idea. I must avoid sync API, especially using `Task.Run` and `GetAwaiter().GetResult()`

File: src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    if (!string.IsNullOrWhiteSpace(content))
+    {
+      return Regex.Replace(content, StripHtmlTagsRegex, string.Empty);
```
Comment (derevnjuk) [inline src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs]: It is not clear why you strip HTML tags here. Why is `VerifySuccessStatusCode` in charge of this?

File: src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+  public static HttpResponseMessage VerifySuccessStatusCode(this HttpResponseMessage httpResponseMessage)
+  {
+    if (httpResponseMessage.CanHandle())
+    {
+      var message = httpResponseMessage.TryExtractMessage();
+
+      if (message is not null)
+      {
+        throw new HttpRequestException(
```
Comment (derevnjuk) [inline src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs]: Why don't introduce a custom exception exposing HTTP status?

**Review by derevnjuk — CHANGES_REQUESTED** [review]
@ostridm I'll take this task on my own

File: src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs (hunk 2 above)
Comment (ostridm) [reply src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs]: At the moment the client receives `html/text`. `IsSuccessStatusCode` is `true` for  `2xx` falimy only.

Comment (derevnjuk) [reply src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs]: So? I can change it to text/plain and your whole logic will be wrong. 

> IsSuccessStatusCode is true for 2xx falimy only.

And this is fine

**Pull request merged**